### DISCUSSION TITLE
[@types/react] TS2322 error after bump version to 18

### DISCUSSION THREAD

**Discussion by akunzai** [DefinitelyTyped/DefinitelyTyped#59759] [Issues with a @types package] [answered]
After Bump @types/react version from 17 to 18,

It throws TS Error: `TS2322: Type 'TFunctionResult' is not assignable to type 'ReactNode'.`

Please take a look [here](https://github.com/akunzai/react-boilerplate/pull/337) for details.

Maybe related to [react-i18next](https://github.com/i18next/react-i18next/issues/1476).

[a comment by an automated account is not shown]

**Comment by eps1lon** [comment]
Could you create a minimal repro for that? Preferably with the [TypeScript playground](https://www.typescriptlang.org/play). 

It seems like this is intended. Some variable is expected to be a `ReactNode` (and therefore be renderable by React). But `TFunctionResult` is probably not renderable. Does it contain some object-like type?

**Reply by akunzai** [reply]
Hi @eps1lon,

The mini [reproduce](https://stackblitz.com/edit/react-ts-otzsat?file=index.tsx).

If I revert the @types/react version to 17.0.44 and refresh, the TS errors gone.

![CleanShot 2022-04-08 at 15 06 50](https://user-images.githubusercontent.com/574719/162382841-26423067-1421-4a85-ab95-a88825c0068a.png)

The `TFunctionResult` can be found at [HERE](https://github.com/i18next/i18next/blob/master/index.d.ts#L685)

the [TFunction](https://github.com/i18next/i18next/blob/master/index.d.ts#L687) should return a string, but it changed to return an object since @types/react v18.

**Reply by eps1lon** [reply]
I mean `t()` can return an `object` according to the types. This would error at runtime as well hence the type-error. If you're sure that `t()` returns a string then you should cast it. Otherwise check to the maintainers of the types of `t` if it is intended that an object is returned.

**Reply by akunzai** [reply]
I got it, thanks for your explanation.

**Reply by alienfacepalm** [reply]
@types/react: "*"

**Comment by akunzai** [comment, marked as answer]
I think this may be caused by https://github.com/DefinitelyTyped/DefinitelyTyped/pull/56026

This issue was resolved by [adding ReactNode to the TDefaultResult type](https://github.com/i18next/react-i18next/pull/1480)
